### PR TITLE
Add export for one Biblio entity.

### PR DESCRIPTION
**Pull request by shushu**
#127

Adding export for one Biblio entity by adding an extra field using hook_field_extra_fields() and biblio.controller.inc buildContent().

export3
export4
selection_011
selection_009
selection_010

### CODE REVIEW

File: modules/biblio_ui/biblio_ui.module [hunk 1]
```diff
@@ -1158,3 +1235,49 @@ function theme_biblio_ui_view_export_area($variables) {
...
+}
+
+/**
+ * Implementing hook_entity_view_alter().
+ * Used to add the Biblio export to the full view of a Biblio entity.
+ *
+ * @param $build
+ * @param $type
+ */
+function biblio_ui_biblio_view_alter(&$build, $type) {
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: I think we should use `hook_field_extra_field()` instead

File: modules/biblio_ui/biblio_ui.module [hunk 2]
```diff
@@ -1158,3 +1235,49 @@ function theme_biblio_ui_view_export_area($variables) {
...
+ */
+function theme_biblio_ui_entity_export_area($variables) {
+  $title = t('Export');
+  $links = array();
+  foreach (biblio_get_exportable_biblio_styles() as $style_name => $style) {
+    $links[$style_name] = array(
+      'href' => 'biblio/export/' . $style_name . '/' . $variables['bid'],
+      'title' => $style['title'],
+      'attributes' => array(
+        'title' => t('Click to download Biblio in @format format', array('@format' => $style_name)),
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module]: Click to download => Download

File: includes/biblio.controller.inc [hunk 3]
```diff
@@ -9,6 +9,20 @@ class Biblio extends Entity {
...
+    if (isset($fields['export']) && $fields['export']['visible']) {
+      $variables = array();
+      $variables['bid'] = $this->bid;
+
+      $content['export'] = array(
+        '#markup' => theme('biblio_ui_entity_export_area', $variables),
+        '#weight' => $fields['export']['weight'],
+      );
+    }
+    return entity_get_controller($this->entityType)->buildContent($this, $view_mode, $langcode, $content);
```
Comment (shushu) [inline includes/biblio.controller.inc:27]: not calling parent.

File: modules/biblio_ui/biblio_ui.module [hunk 4]
```diff
@@ -1158,3 +1235,48 @@ function theme_biblio_ui_view_export_area($variables) {
...
+  $attributes = array(
+    'links' => $links,
+    'title' => $title,
+    'attributes' => array('class' => array('biblio-export-buttons')),
+  );
+  return theme('ctools_dropdown', $attributes);
+}
+
+/**
+ * Implementation of hook_field_extra_fields().
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module:1264]: Implementation of => Implements

File: modules/biblio_ui/biblio_ui.module [hunk 5]
```diff
@@ -1158,3 +1235,48 @@ function theme_biblio_ui_view_export_area($variables) {
   );
   return theme('ctools_dropdown', $attributes);
 }
+
+/**
+ * Theme callback - display a button for export in Biblio entity.
```
Comment (amitaibu) [inline modules/biblio_ui/biblio_ui.module:1240]: `Theme callback -` => `Theme callback;`

**Comment by shushu** [comment]
Merged with PR #139

**Pull request closed without merging**